File: src/app/lib/src/dtos/input_models/add_message_to_chat_input_model.dart
import "dart:convert";

class AddMessageToChatInputModel {
  final String? chatId;
  final String messageId;
  final List<int> participants;
  final String content;

  AddMessageToChatInputModel({
    this.chatId,
    required this.messageId,
    required this.participants,
    required this.content
  });

  String toJson() => json.encode({
    "chatId": this.chatId,
    "messageId": this.messageId,
    "participants": this.participants,
    "content": this.content
  });
}

File: src/API/Microsservices/Chat/Sonorus.Chat.Application/Commands/AddMessageToChat/AddMessageToChatInputModel.cs
namespace Sonorus.Chat.Application.Commands.AddMessageToChat;

public class AddMessageToChatInputModel(string? chatId, string? messageId, IEnumerable<long> participants, string content) {
    public string? ChatId { get; private set; } = chatId;
    public string? MessageId { get; private set; } = messageId;
    public IEnumerable<long> Participants { get; private set; } = participants;
    public string Content { get; private set; } = content;
}